lib/service/task_service.dart

```diff
@@ -20,27 +21,60 @@ class TaskService {
     }
   }
 
-  Future<List<Task>> readTasks() async {
-    dynamic data = await client.getData(uri: taskApiBase);
-    List<Task> tasks =
-        List<Task>.from(data.map((taskData) => Task.fromJson(taskData)));
-    print(data);
-    return tasks;
-  }
-
-  Future<void> editTask({required Task task}) async {
+  Future<List<Task>> readTasksByStatus(TaskStatus status) async {
     try {
-      await client.putData(
-        uri: '$taskApiBase/${task.id}',
-        body: task.toJson(),
+      dynamic data = await client.getData(
+        uri: '/tasks',
+        queryParams: {'status': Task.getStatusString(status)},
       );
+
+      if (data is List) {
+        List<Task> tasks =
+            List<Task>.from(data.map((taskData) => Task.fromJson(taskData)));
+        return tasks;
+      } else {
+        print('Unexpected response format: $data');
+        return [];
+      }
     } catch (e) {
-      print('Error editing task: $e');
-      throw Exception('Edit Task Failed');
+      print('Error fetching tasks: $e');
+      return [];
     }
   }
 
-  Future<void> completeTask({required Task task}) async {}
+  Future<void> updateTask({
+    required Task task,
+    TaskStatus? newStatus,
+  }) async {
+    try {
+      if (newStatus != null) {
+        task = task.copyWithStatus(newStatus);
+        await client.putData(
+          uri: '$taskApiBase/${task.id}/${Task.getStatusString(newStatus)}',
+          body: task.toJson(),
+          queryParams: {'status': Task.getStatusString(newStatus)},
```

Comment:
I'm not sure why you have the query parameter or the status in the URI. Doesn't the updated status come from the `body`? I see the `newStatus` sent to the back-end three times.

Reply:
You're right, I think I was running into issues because in my test coverage it was saying this block wasn't being reached, and they only way I could fix it was by adding the query params. I edited this, and tweaked the tests a bit, and then checked the coverage and it was fine. Not sure why I was running into those issues before but this helped. Thank you!

Reply:
in fact, it might have boosted my coverage by a % or two